internal/cloud/realm/app.go

```diff
@@ -0,0 +1,137 @@
+package realm
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"net/http"
+
+	"github.com/10gen/realm-cli/internal/utils/api"
+)
+
+// UserProfile contains all of the roles associated with a user
+type UserProfile struct {
+	Roles []Role `json:"roles"`
+}
+
+// Role contains a GrouID field that maps to a project ID
+type Role struct {
+	GroupID string `json:"group_id"`
+}
+
+// App represents basic Realm App data
+type App struct {
+	ID          string `json:"_id"`
+	GroupID     string `json:"group_id"`
+	ClientAppID string `json:"client_app_id"`
+	Name        string `json:"name"`
+}
+
+var (
+	errGroupNotFound = errors.New("group could not be found")
+)
+
+var (
+	appsByGroupIDEndpoint = adminAPI + "/groups/%s/apps"
+	userProfileEndpoint   = adminAPI + "/auth/profile"
+)
+
+// FilterGroupIDsFromUserProfile returns all available group ids for a given user
+func (pd *UserProfile) FilterGroupIDsFromUserProfile(projectFlag string) []string {
+	groupIDs := []string{}
+	set := make(map[string]bool)
+	for _, role := range pd.Roles {
+		// Break early if we find the groupID that matches the project flag
+		if len(projectFlag) == 0 && !set[role.GroupID] {
+			groupIDs = append(groupIDs, role.GroupID)
+			set[role.GroupID] = true
+			continue
+		}
+
+		if role.GroupID == projectFlag {
+			return []string{role.GroupID}
+		}
+	}
+
+	return groupIDs
+}
+
+func (c *client) GetUserProfile() (UserProfile, error) {
+	opts, _ := api.JSONRequestOptions(nil)
+
+	res, resErr := c.do(http.MethodGet, appsByGroupIDEndpoint, opts)
+	if resErr != nil {
+		return UserProfile{}, resErr
+	}
+
+	if res.StatusCode != http.StatusOK {
+		return UserProfile{}, UnmarshalServerError(res)
+	}
+
+	dec := json.NewDecoder(res.Body)
+	defer res.Body.Close()
+
+	var profile UserProfile
+	if err := dec.Decode(&profile); err != nil {
+		return UserProfile{}, err
+	}
+
+	return profile, nil
+}
+
+func (c *client) FindProjectAppByClientAppID(groupIDs []string, appFlag string) ([]App, error) {
```

Comment:
for the client, let's define/implement the following two methods for your purposes here:

```go
func (c *client) GetAppsByUser(groupIDs []string) ([]App, error) { /* iterate over groupIDs and collect FindAppsByGroupID */ }
func (c *client) GetAppsByGroupID(groupID string) ([]App, error) { /* lets fail early for now if anything goes wrong here */ }
// TODO: just for demo purposes, you won't need this one yet, no need to implement
func (c *client) GetApp(groupID, clientAppID string) (App, error) { /* ... */ }
```

this should be all you need on the `realm/client` side to get `app list` running

Reply:
Done! Commented out the tests that previously checked app id (though I can just remove if preferred). I have an if statement in the app list handler that calls GetApps if the project flag is provided. Let me know if that was not the intended solution here but I imagine that the same logic would be applied with the app flag as well